test/plugin/test_in_tcp.rb

```diff
@@ -226,6 +226,8 @@ def create_tcp_socket(host, port, &block)
 
       assert_equal 1, d.instance.log.logs.count { |l| l =~ /anonymous client/ }
       assert_equal 0, d.events.size
+
+      d.instance_shutdown
```

Comment:
Indeed. Need it since the driver ran with `shutdown: false`.

Reply:
I think we shouldn't use `shutdown: false` as possible although I'm not sure whether it's needed here or not.
FYI sometimes I found some needless `shutdown: false` in other tests (probably copy&pasted).

Reply:
When terminating a plugin on shutdown, the `reset` of the logger is called and it erases the logs.

So this test sets `shutdown: false`.

But somehow the data of `driver.logs` is still alive after terminating...

Reply:
So I think using `driver.logs` is the correct way, but I need more time to check how it works.

Reply:
Is it better to fix it in this PR?
Or should I fix it in another PR?

Reply:
It's fine as is (`shutdown: false`).
Instead it should be run in ensure block to make sure to stop:
```suggestion
  ensure
      d.instance_shutdown if d && d.instance
```

Reply:
Now I understand this.
I made an issue.

* #4136

Reply:
I made a PR

* #4138